tests/conftest.py

```diff
@@ -396,3 +397,56 @@ def log_cli_level(pytestconfig):
 
     """
     return pytestconfig.getini('log_cli_level') or 'DEBUG'
+
+
+@pytest.fixture(scope="class")
+def run_io_in_background(request):
+    """
+    Run IO during the test execution
+    """
+    if config.RUN['cli_params'].get('io_in_bg'):
+
+        lock = threading.Lock()
+
+        def finalizer():
+            """
+            """
+            with lock:
+                request.test_running = [False]
+            thread.join()
+            for result in results:
+                log.info(f"IOPs after FIO for pod {pod_obj.name}:")
+                log.info(
+                    f"Read: {result.get('jobs')[0].get('read').get('iops')}"
+                )
+                log.info(
+                    f"Write: {result.get('jobs')[0].get('write').get('iops')}"
+                )
+            pod_obj.delete()
+            pod_obj.ocp.wait_for_delete(resource_name=pod_obj.name)
+            pvc_obj.delete()
+            pvc_obj.ocp.wait_for_delete(resource_name=pvc_obj.name)
+
+        request.addfinalizer(finalizer)
+
+        pvc_obj = helpers.create_pvc(sc_name=defaults.DEFAULT_STORAGECLASS_NAME, size='2Gi')
+        pod_obj = helpers.create_pod(interface_type=constants.CEPHBLOCKPOOL, pvc_name=pvc_obj.name)
+
+        file_path = os.path.join(pod_obj.get_mount_path(), 'fio-rand-readwrite')
+
+        with lock:
+            request.test_running = [True]
+
+        results = list()
+
+        def run_io_in_bg(state):
+            while True:
+                with lock:
+                    if state[0]:
+                        pod_obj.run_io('fs', '1G')
```

Comment:
Can you please let me know how and where the data integrity is checked?

Reply:
It is not checked. What I had in mind here is to simply have a load in the BG while the tests are running, to simulate a real-world scenario